server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -164,13 +164,15 @@ public String executor() {
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
             final String extensionTokenProcessor = "placeholder_token_processor";
             final String requestIssuerIdentity = "placeholder_request_issuer_identity";
+            // TODO Token is currently generated in SecurityRestFilter of Identity module. Should that be moved here?
+            String jwt = client.threadPool().getThreadContext().getHeader("_opensearch_auth_token");
 
             transportService.sendRequest(
                 discoveryExtensionNode,
                 ExtensionsManager.REQUEST_REST_EXECUTE_ON_EXTENSION_ACTION,
                 // HERE BE DRAGONS - DO NOT INCLUDE HEADERS
                 // SEE https://github.com/opensearch-project/OpenSearch/issues/4429
-                new ExtensionRestRequest(method, path, params, contentType, content, requestIssuerIdentity),
+                new ExtensionRestRequest(method, path, params, contentType, content, jwt),
```

Comment:
will this jwt replace `requestIssuerIdentity` which was the request principal?

Reply:
okay..this will serve as an identifier of the requester, correct? or will it identify extension too?

Reply:
That's an area that hasn't been looked into yet. The token is signed by the secret for the extension so when the extension tries to interact with the cluster via REST calls, the token is passed as an `Aurthorization: Bearer <jwt>` and is verified using the secret for the extension. atm, I am not sure how to identify which extension a call is coming from.